pkg/backend/display/progress.go

```diff
@@ -406,11 +400,24 @@ func (r *CaptureProgressEvents) ProcessEvents(
 	renderChan <-chan engine.Event,
 	renderDone chan<- bool,
 ) {
+	// Reuse the shared processing logic
```

Comment:
I don't think this and the below comment are particularly useful to understand the code in the future. Not sure they are worth adding.